mindsdb_native/libs/data_types/transaction_data.py

```diff
@@ -4,13 +4,26 @@
 class TransactionData:
     def __init__(self):
         self.data_frame = None
-        self.train_df = None
-        self.test_df = None
-        self.validation_df = None
         self.columns = []
 
         self._sample_df = None
 
+        self.train_idx = None
+        self.validation_idx = None
+        self.test_idx = None
+
+    @property
+    def train_df(self):
+        return self.data_frame.loc[self.train_idx]
```

Comment:
This change seems like a beyond horrible idea to me.

Index Accessing in pandas is slow and these will be accessed .... hundreds, thousands, millions of times.

This should be removed at all cost, I would be fine with it *IF* it had a cache but even then it seems like a fairly pointless change and I'd rather not change the code unless there's a good reason to.

But maybe I'm wrong here.